scripts/find_unique_isoform.py

```diff
@@ -128,64 +172,83 @@ def compareFunc(seqDict, outFile, minPercent):
                          'isoform_sequence', \
                          'selected_alignments']) +'\n')
 
-    for locus,allSamples in seqDict.items():
-
-        # all isoforms of the locus
-        '''allSeqs, allNames = [], []
-        for s,d in allSamples.items():
-            allSeqs += [ d[q]['seq'] for q in d.keys() ]
-            allNames += [ s+'_'+q for q in d.keys() ]
-        total = len(allSeqs)
-        '''
-        total = sum([ len([ allSamples[sample][qname]['seq'] for qname in allSamples[sample].keys() ]) for sample in allSamples.keys() ])
+    outDiver.write('\t'.join(['win_chr', \
+                         'win_start', \
+                         'win_end', \
+                         'total_isoform', \
+                         'isoform_name', \
+                         'sample_from', \
+                         'mapped_start', \
+                         'identical_isoforms', \
+                         'divergent_isoforms']) +'\n')
 
-        lookup = {}
+    for locus,isoforms in seqDict.items():
 
-        for sample,sampleDict in allSamples.items():
+        total = len(isoforms)
+        samples = sorted(list( set([ s.sample for s in isoforms ]) ))
 
-            # isoforms in this sample
-            this = [ sampleDict[q]['seq'] for q in sampleDict.keys() ]
+        lookup = {}
 
-            for anotherSample in allSamples.keys():
+        # unique isoforms
+        for sample in samples:
+            this = [ s.seq for s in isoforms if s.sample == sample ]
```

Comment:
Variable names could be more informative. 
e.g. 
this -> this_sample_isoforms
another -> another_sample_isoforms

Reply:
+1 in agreement with the above

I followed the code, but a more informative variable name would have saved me a few seconds of thought --> good